--- a/src/switch_core_io.c	2018-01-24 10:54:30.000000000 -0800
+++ b/src/switch_core_io.c	2019-01-17 11:21:20.780559360 -0800
@@ -1459,7 +1459,7 @@
 				switch_core_media_bug_flush_all(session);
 			}
 
-			if (!(switch_buffer_write(session->raw_write_buffer, write_frame->data, write_frame->datalen))) {
+			if (write_frame->datalen > 0 && (!(switch_buffer_write(session->raw_write_buffer, write_frame->data, write_frame->datalen)))) {
 				switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_ERROR, "Write Buffer %u bytes Failed!\n", write_frame->datalen);
 				status = SWITCH_STATUS_MEMERR;
 				goto error;
